File: dotnet/progressive/Models/Batch.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System;

namespace progressive.Models
{
    public class Batch
    {
        // [DatabaseGenerated(DatabaseGeneratedOption.None)]
        public int BatchID { get; set; }
        public string BatchCode { get; set; }
        public DateTime Term1StartDate { get; set; }
        public DateTime Term2StartDate { get; set; }
        public DateTime MidTerm1Date { get; set; }
        public DateTime EndTerm1Date { get; set; }
        public DateTime MidTerm2Date { get; set; }
        public DateTime EndTerm2Date { get; set; }
        public bool IsActive {get; set; }

        public ICollection<Group> Groups { get; set; }
    }
}

File: angular/progression/src/app/models/Batch.ts
import { Group } from './Group';


export interface Batch {
  batchId: number;
  batchCode: string;
  term1StartDate: Date | string;
  term2StartDate: Date | string;
  midTerm1Date: Date | string;
  endTerm1Date: Date | string;
  midTerm2Date: Date | string;
  endTerm2Date: Date | string;
  isActive: boolean;

  groups: Group[];
}

